docs/getting-started/learn-the-api.md

```diff
@@ -131,33 +131,119 @@ const isAuthenticated = useIsAuthenticated();
 
 The session is also available from the [Runtime](/reference/runtime/runtime-class.md) instance.
 
-## Services
+<!-- ## Plugins
 
-While Squide provides a range of built-in functionalities, by no mean these alone can support the needs of every mature application. Therefore, the shell [Runtime](/reference/runtime/runtime-class.md) allows the addition of custom services.
+To keep Squide lightweight, not all functionalities should be integrated as a core functionality. However, to accommodate a broad range of technologies, a plugin system has been implemented to fill the gap.
 
-First, make the service available to every part of the application by passing a service instance to the `Runtime` instance:
+First, define a plugin by implementing the [Plugin](../reference/plugins/plugin.md) interface:
+
+```ts shared/src/mswPlugin.ts
+import { Plugin } from "@squide/react-router";
+import type { RestHandler } from "msw";
+
+export class MswPlugin extends Plugin {
+    constructor() {
+        super(MswPlugin.name);
+    }
+
+    registerRequestHandlers(handlers: RestHandler[]) {
+        ...
+    }
+}
+```
+
+Then, make the plugin available to every part of the application by passing a service instance to the `Runtime` instance:
 
 ```ts host/src/boostrap.tsx
 import { Runtime } from "@squide/react-router";
-import { UserService } from "@sample/shared";
+import { MswPlugin } from "@squide/msw";
 
 const runtime = new Runtime({
-    services: {
-        "user-service": new UserService()
-    }
+    plugins: [new MswPlugin()]
 });
 ```
 
-Then, access the service instance from anywhere with the [useService](/reference/runtime/useService.md) hook:
+Then, access the plugin instance from the [Runtime](/reference/runtime/runtime-class.md) instance:
 
 ```ts
-import { useService } from "@squide/react-router";
-import { type UserService } from "@sample/shared";
+import { MswPlugin } from "@sample/shared";
+import { requetHandlers } from "../mocks/handlers.ts";
+
+const mswPlugin = runtime.getPlugin(MswPlugin.name) as MswPlugin;
+``` -->
+
+## Mock Service Worker
 
-const service = useService("user-service") as UserService;
+We recommend to mock the API endpoints with [Mock Service Worker](https://mswjs.io/) (MSW) to faciliate the development and encourage an [Contract Design First](https://devblogs.microsoft.com/ise/2023/05/08/design-api-first-with-typespec/) approach.
+
+To help with that, a `@squide/msw` plugin is available.
```

Comment:
```suggestion
To help with that, a `@squide/msw` package is available.
```

Reply:
Or rework the phrase so it say a plugin MswPlugin is available in the new @squide/msx package